src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -2,32 +2,124 @@
 
 import java.util.List;
 
-public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+class Node<T> {
+    T data;
+    Node<T> prev;
+    Node<T> next;
```

Comment:
The fields `data`, `prev`, and `next` of the `Node` class should be private to follow encapsulation principles.